sdk/core/core-amqp/src/errors.ts

```diff
@@ -560,6 +553,19 @@ export function translate(err: AmqpError | Error): MessagingError {
 
   let error: MessagingError = err as MessagingError;
 
+  // OperationTimeoutError is retryable as it indicates potential problem with unexpected delays
+  // and not an issue with the Messaging process
+  if (
+    err instanceof OperationTimeoutError ||
+    // instanceof checks on custom Errors doesn't work without manually setting the prototype within the error.
```

Comment:
Since instanceOf checks don't work on custom errors as documented here, then why do we have the instanceOf check here at all?

Reply:
Per documentation, this was to make sure the checks are compatible and resilient to version changes right?

Reply:
I think checking the `name` field on Error is more resilient than doing an `instanceof` check. I don't know how `OperationTimeoutError` is constructed, but the comment suggests that the `instanceof` check will always fail. I agree with @ramya-rao-a that if the check is always going to return `false`, then it shouldn't be needed.

Reply:
On checking the implementation of the error, I found that the prototype was not being set as seems needed. Hence, assumed that this holds true. Was there anything else about the construction of error to look at such?

Reply:
Ah, looks like I misunderstood the intended changes, as pointed name check is resilient/complete and not an additional requirement - removed the extraneous check.